### DISCUSSION TITLE
Can I exclude part of codes in one file, or just check the increase code in gitlab mr?

### DISCUSSION THREAD

**Discussion by liyuzero** [detekt/detekt#6705] [Q&A] [answered]
now, detekt will check the whole content of modified file, but not just check the increased code, is there anyway to just check changed code in a modified file?

**Comment by schalkms** [comment, marked as answer]
Short answer: no

Longer answer:
With rules using type and symbol solving, detekt will need the information from referenced types as well. For more sophisticated analysis, detekt needs more context. Using just parts of changed files is in this case not enough.

Questions back:
Why would you want static analysis just over changed lines of code?
